src/snippet/mod.rs

```diff
@@ -186,6 +186,29 @@ fn select_best_fragment_combination(fragments: &[FragmentCandidate], text: &str)
     }
 }
 
+/// Returns ranges that are collapsed into non-overlapped ranges.
+fn collapse_overlapped_ranges(ranges: &Vec<Range<usize>>) -> Vec<Range<usize>> {
+    let mut result = Vec::new();
+    let mut ranges = ranges.iter();
```

Comment:
nitpick... I'm not fond of shadowing. 
It is sometime useful to prevent the usage of a variable.

Here I would either call it `ranges_it`, or pass an Iterator as argument.